src/Lykke.Service.PayInternal.Services/PaymentRequestDetailsBuilder.cs

```diff
@@ -1,14 +1,46 @@
-﻿using System.Threading.Tasks;
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Lykke.Service.PayInternal.Core.Domain.Order;
 using Lykke.Service.PayInternal.Core.Domain.PaymentRequests;
+using Lykke.Service.PayInternal.Core.Domain.Transaction;
 using Lykke.Service.PayInternal.Core.Services;
 
 namespace Lykke.Service.PayInternal.Services
 {
     public class PaymentRequestDetailsBuilder : IPaymentRequestDetailsBuilder
     {
-        public async Task<PaymentRequestDetails> Build(IPaymentRequest paymentRequest)
+        private readonly IOrderService _orderService;
+        private readonly ITransactionsService _transactionsService;
+
+        public PaymentRequestDetailsBuilder(
+            IOrderService orderService,
+            ITransactionsService transactionsService)
+        {
+            _orderService = orderService;
+            _transactionsService = transactionsService;
+        }
+
+        public async Task<TResult> Build<TResult, TOrder, TTransaction, TRefund>(IPaymentRequest paymentRequest,
+            PaymentRequestRefund refundInfo)
         {
-            throw new System.NotImplementedException();
+            IOrder order = await _orderService.GetAsync(paymentRequest.Id, paymentRequest.OrderId);
+
+            IReadOnlyList<IPaymentRequestTransaction> transactions =
+                await _transactionsService.GetByWalletAsync(paymentRequest.WalletAddress);
+
+            IEnumerable<IPaymentRequestTransaction> paymentTransactions = transactions.Where(x => x.IsPayment());
+
+            dynamic model = Mapper.Map<TResult>(paymentRequest);
```

Comment:
Who is consuming this dynamic model?

Reply:
Nobody,
It is just to be able to assemble the single object from multiple objects (order, transactions, refund info etc)